src/core/MOM.F90

```diff
@@ -3051,16 +3083,27 @@ subroutine extract_surface_state(CS, sfc_state_in)
   if (CS%rotate_index) &
     turns = G%HI%turns
 
-  if (.not.sfc_state_in%arrays_allocated) &
+  if (.not.sfc_state_in%arrays_allocated)  then
     !  Consider using a run-time flag to determine whether to do the vertical
     ! integrals, since the 3-d sums are not negligible in cost.
-    call allocate_surface_state(sfc_state_in, G_in, use_temperature, &
-        do_integrals=.true., omit_frazil=.not.associated(CS%tv%frazil))
+    if (associated(CS%frac_shelf_h)) then
```

Comment:
This could be coded more parsimoniously here and at line 3101 by introducing a logical variable `use_iceshelves = associated(CS%frac_shelf_h)` that is passed to what would become 2 calls (instead of 4) to `allocate_surface_state()`.